src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -52,11 +66,13 @@ public KafkaMonitor(Map<String, Map> testProps) throws Exception {
         _services.put(name, service);
       }
     }
-
-    _executor = Executors.newSingleThreadScheduledExecutor();
   }
 
-  public void start() {
+  public synchronized void start() {
+    if (_isShutdown) {
```

Comment:
If the motivation of this check is to prevent `start()` from being invoked more than once, should we do something similar to `ProduceService.start()`?